### PR TITLE
36 new function calculate polygon centroid

### LINKED ISSUE #24: [NEW function] CRS_extracter.R
## Functionality
A function that returns the correct crs string for spatial operations

## Checklist
- [x] copy [CRS_extracter.R](https://github.com/inbo/backoffice-wild-analyse/blob/master/Functies/CRS_extracter.R) from the `backoffice` to `./R`
- [x] voorzie een functie titel met `#'` op regel 1 van je script
- [x] voorzie een auteur met `#' @author`
- [x] voorzie een beschrijving met `#' @description`
- [x] voorzie uitleg over de input parameter(s) met `#' @param name`
- [x] voorzie uitleg over de output van de functie met `#' @returns`
- [x] voorzie minstens 1 voorbeeld van het gebruik van de functie dmv `#' @examples`
- [x] voorzie de export - tag dmv `#' @export` ([#14](https://github.com/inbo/fistools/issues/14))
- [x] voer `usethis::use_package("packagename", min_version = TRUE)` uit in de console voor iedere package die je gebruikt. 
- [x] voer `roxygen2::roxygenise()` uit in de console
- [x] voer `devtools::check()` uit in de console
- [x] los eventuele errors, warnings en notes<sup>1</sup> op
- [x] maak een pull request met @soriadelva of @SanderDevisscher en eventueel andere relevante gebruikers als reviewer.

*<sup>1</sup>in de mate van het mogelijke*

### LINKED ISSUE #36: [NEW function] calculate_polygon_centroid
## Functionality
- [x] creates a dataframe with `centroidLatitude`, `centroidLongitude` & `coordinateUncertaintity`

## Checklist
- [x] maak een nieuw R-bestand 
- [x] sla het R bestand op onder `./R` met filenaam is gelijk aan functienaam
- [x] voorzie een functie titel met `#'` op regel 1 van je script
- [x] voorzie een auteur met `#' @author`
- [x] voorzie een beschrijving met `#' @description`
- [x] voorzie uitleg over de input parameter(s) met `#' @param name`
- [x] voorzie uitleg over de output van de functie met `#' @returns`
- [x] voorzie minstens 1 voorbeeld van het gebruik van de functie dmv `#' @examples`
- [x] voorzie de export - tag dmv `#' @export` ([#14](https://github.com/inbo/fistools/issues/14))
- [x] voer `usethis::use_package("packagename", min_version = TRUE)` uit in de console voor iedere package die je gebruikt. 
- [x] voer `roxygen2::roxygenise()` uit in de console
- [x] voer `devtools::check()` uit in de console
- [x] los eventuele errors, warnings en notes<sup>1</sup> op
- [x] increment versie dmv `usethis::use_version(which = "minor")`<sup>2</sup>
- [x] maak een pull request met @soriadelva of @SanderDevisscher en eventueel andere relevante gebruikers als reviewer.

*<sup>1</sup>in de mate van het mogelijke*

*<sup>2</sup>als je meer dan 5 functies en/of datasets in één pull request veranderd hebt gebruik: `usethis::use_version(which = "major")`*

### PR DESCRIPTION
**Pull request by SanderDevisscher**
Creates the following functions:

Fixes #24 

- install_sp: Helper function that installs sp when missing, from a tarball.
- CRS_extracter: Extracts a coordinate reference system (CRS) from a library of commonly used CRS codes
- lib_crs.rda: library of commonly used CRS codes
- imports some functions and classes from sp & utils

Fixes #36 

- Calculate the centroid of a polygon: Calculates the centroid latitude & longitude and the uncertainty of this coordinate.

### CODE REVIEW

File: R/CRS_extracter.R [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  if(grepl("wgs", CRS, ignore.case = TRUE)){
+    CRS <- "WGS"
+  }
+
+  if(grepl("bel", CRS, ignore.case = TRUE)){
+    CRS <- "BEL72"
+  }
+
+  if(EPSG == TRUE){
+    CRS_output <- CRS(Lib_CRS$EPSG[CRS == Lib_CRS$CRS_Naam])
```
[the remark this answers is not in the document]
Comment (SanderDevisscher) [reply R/CRS_extracter.R:59]: did you do devtools::load_all() prior to running code ?

[the remark this answers is not in the document]
Comment (SanderDevisscher) [reply R/CRS_extracter.R:59]: the function CRS is imported from "sp"

File: R/CRS_extracter.R [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }else{
```
Comment (soriadelva) [inline R/CRS_extracter.R:60]: When running this part of the script, I get the following error message:
`Error in CRS(Lib_CRS$EPSG[CRS == Lib_CRS$CRS_Naam]) :  could not find function "CRS". `
I'll have to do the review of the polygon centroid function after this is fixed.

**Review by soriadelva — CHANGES_REQUESTED** [review]

File: R/CRS_extracter.R (hunk 2 above)
Comment (soriadelva) [reply R/CRS_extracter.R:60]: I did do devtools::load_all() but it yielded the following error message:
` ℹ The package "osmdata" (>= 0.2.5) is required.
✖ Would you like to install it?

1: Yes
2: No

Selection: 1
Error: 
! error in pak subprocess
Caused by error in get_remote_types(refs):
! Cannot parse package: C:/Users/soria_delva/Documents/GitHub/fistools.
ℹ See ?pkgdepends::pkg_refs() for supported package sources.
Type .Last.error to see the more details. `

Comment (SanderDevisscher) [reply R/CRS_extracter.R:60]: try `install.packages("osmdata")`

Comment (soriadelva) [reply R/CRS_extracter.R:60]: It got fixed after I installed the osmdata package independently. There seemed to be a problem when it was installed in the way indicated above

Comment (soriadelva) [reply R/CRS_extracter.R:60]: And that fixed the error in CRS_extracter 👍

Comment (SanderDevisscher) [reply R/CRS_extracter.R:60]: yeah 😎

**Review by soriadelva — APPROVED** [review]

File: R/CRS_extracter.R (hunk 2 above)
Comment (soriadelva) [reply R/CRS_extracter.R:60]: function works like a charm 🧙

**Pull request merged**